src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs

```diff
@@ -128,6 +128,18 @@ private static bool TryExtractPlatformName(string methodName, [NotNullWhen(true)
                     osPlatformName = propertyReference.Property.Name;
                     return true;
                 }
+                else if (argumentValue is IInvocationOperation iOperaion &&
+                    iOperaion.TargetMethod.ReturnType.Equals(osPlatformType) &&
+                    iOperaion.Arguments.Length == 1 &&
+                    iOperaion.Arguments[0] is { } argument &&
+                    argument.Value is ILiteralOperation literal &&
+                    literal.Type.SpecialType == SpecialType.System_String &&
+                    literal.ConstantValue.HasValue &&
+                    !literal.ConstantValue.Value.Equals(string.Empty))
+                {
+                    osPlatformName = literal.ConstantValue.Value.ToString();
```

Comment:
If you are going for the previous change, you could simply do

```suggestion
                    osPlatformName = platformName;
```